util/src/main/java/tc/oc/pgm/util/nms/packets/BlockEntity.java

```diff
@@ -0,0 +1,14 @@
+package tc.oc.pgm.util.nms.packets;
+
+import org.bukkit.Material;
+import org.bukkit.entity.Entity;
+
+public interface BlockEntity {
+    boolean isDisplayEntity();
+    Entity entity();
```

Comment:
ideally you should NOT expose the entity itself, that there even is an entity is an implementation detail that should be hidden from the observer, because if for example armor stands are used, the position of the armor stand itself will not align with the position of the block entity (the armor stand feet will always be 1.8 blocks below), so if you expose the entity, that illusion breaks.

if instead you add here move methods, an armor-stand-backed impl can:
```java
private static final double OFFSET = 1.8f;

void teleport(Location loc) {
  loc = loc.clone();
  loc.setY(loc.getY() - OFFSET);
  this.entity.teleport(loc);
}
```

which is the whole idea behind hiding this in impl-dependant methods